### PR TITLE
[VL] Support PreciseTimestampConversion function

### PR DESCRIPTION
**Pull request by zhli1142015**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by zhli1142015** [comment]
cc @rui-mo  and @PHILO-HE , thanks.

File: backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala [hunk 1]
```diff
@@ -1077,4 +1077,25 @@ class ScalarFunctionsValidateSuite extends FunctionsValidateTest {
...
+              "('A2', TIMESTAMP'2021-01-01 00:01:00') AS tab(a, b)")
+          .write
+          .parquet(path.getCanonicalPath)
+
+        spark.read.parquet(path.getCanonicalPath).createOrReplaceTempView("string_timestamp")
+
+        runQueryAndCompare(
+          "SELECT a, window.start, window.end, count(*) as cnt FROM" +
+            " string_timestamp GROUP by a, window(b, '5 minutes') ORDER BY a, start;") {
+          checkGlutenOperatorMatch[ProjectExecTransformer]
```
Comment (rui-mo) [inline backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala:1097]: Is the round-trip conversion tested in this test?

**Review by rui-mo — APPROVED** [review]
Thanks.

File: backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala (hunk 1 above)
Comment (zhli1142015) [reply backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala:1097]: Yes, below is the fallback before this change.
```
(9) Project [codegen id : 2]
Output [2]: [named_struct(start, knownnullable(precisetimestampconversion(((precisetimestampconversion(b#28, TimestampType, LongType) - CASE WHEN (((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) < 0) THEN (((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) + 300000000) ELSE ((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) END) - 0), LongType, TimestampType)), end, knownnullable(precisetimestampconversion((((precisetimestampconversion(b#28, TimestampType, LongType) - CASE WHEN (((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) < 0) THEN (((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) + 300000000) ELSE ((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) END) - 0) + 300000000), LongType, TimestampType))) AS window#33, a#27]
Input [2]: [a#27, b#28]
```

**Comment by jinchengchenghh** [comment]
This test cannot test if the function is used.
Because `knownnullable` is not supported, so the function fallback, it just check `(8) ProjectExecTransformer [a#143, window#169.start AS start#170, window#169.end AS end#171, count(1)#168L AS cnt#167L]`
The plan is 
```
VeloxColumnarToRowExec
   +- ^(9) SortExecTransformer [a#143 ASC NULLS FIRST, start#170 ASC NULLS FIRST], true, 0
      +- ^(9) InputIteratorTransformer[a#143, start#170, end#171, cnt#167L]
         +- ShuffleQueryStage 1
            +- ColumnarExchange rangepartitioning(a#143 ASC NULLS FIRST, start#170 ASC NULLS FIRST, 1), ENSURE_REQUIREMENTS, [plan_id=633], [id=#633], [OUTPUT] List(a:StringType, start:TimestampType, end:TimestampType, cnt:LongType), [OUTPUT] List(a:StringType, start:TimestampType, end:TimestampType, cnt:LongType)
               +- ^(8) ProjectExecTransformer [a#143, window#169.start AS start#170, window#169.end AS end#171, count(1)#168L AS cnt#167L]
                  +- ^(8) HashAggregateTransformer(keys=[a#143, window#169], functions=[count(1)], output=[a#143, window#169, count(1)#168L])
                     +- ^(8) InputIteratorTransformer[a#143, window#169, count#179L]
                        +- RowToVeloxColumnar
                           +- ShuffleQueryStage 0
                              +- Exchange hashpartitioning(a#143, window#169, 1), ENSURE_REQUIREMENTS, [plan_id=567]
                                 +- VeloxColumnarToRowExec
                                    +- ^(7) FlushableHashAggregateTransformer(keys=[a#143, window#169], functions=[partial_count(1)], output=[a#143, window#169, count#179L])
                                       +- ^(7) InputIteratorTransformer[window#169, a#143]
                                          +- RowToVeloxColumnar
                                             +- *(2) Project [named_struct(start, knownnullable(precisetimestampconversion(((precisetimestampconversion(b#144, TimestampType, LongType) - CASE WHEN (((precisetimestampconversion(b#144, TimestampType, LongType) - 0) % 300000000) < cast(0 as bigint)) THEN (((precisetimestampconversion(b#144, TimestampType, LongType) - 0) % 300000000) + 300000000) ELSE ((precisetimestampconversion(b#144, TimestampType, LongType) - 0) % 300000000) END) - 0), LongType, TimestampType)), end, knownnullable(precisetimestampconversion((((precisetimestampconversion(b#144, TimestampType, LongType) - CASE WHEN (((precisetimestampconversion(b#144, TimestampType, LongType) - 0) % 300000000) < cast(0 as bigint)) THEN (((precisetimestampconversion(b#144, TimestampType, LongType) - 0) % 300000000) + 300000000) ELSE ((precisetimestampconversion(b#144, TimestampType, LongType) - 0) % 300000000) END) - 0) + 300000000), LongType, TimestampType))) AS window#169, a#143]
                                                +- VeloxColumnarToRowExec
                                                   +- ^(6) FilterExecTransformer isnotnull(b#144)
                                                      +- ^(6) InputIteratorTransformer[a#143, b#144]
                                                         +- RowToVeloxColumnar
                                                            +- *(1) ColumnarToRow
                                                               +- BatchScan parquet file:/tmp/spark-3ebf1be6-44e8-4f33-a661-3540f556
```

**Comment by zhli1142015** [comment]
> `knownnullable`

Thanks for reproting this. I think `KnownNullable` is already supported by https://github.com/apache/incubator-gluten/pull/5365/files, but not portable to spark 3.5, let me fix it and also test.

**Comment by jinchengchenghh** [comment]
Thanks!

**Pull request merged**